### PR TITLE
Moved common SQL handler methods of common-sql-provider into dedicated module

### PR DESCRIPTION
**Pull request by dabla**
This PR is linked to the original [PR Introduce notion of dialects in DbApiHook](https://github.com/apache/airflow/pull/41327), but only contains a small refactoring in which the sql handlers of the common-sql -provider are moved from the sql module to a dedicated handler module, as this will be needed later on to avoid circular import issues. This is to make the review easier as asked by @potiuk.  This PR only impacts the common-sql provider, also a dedicated unit test has been added for the handlers.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by dabla** [comment]
Some things are failing but dunno why

**Comment by potiuk** [comment]
It seems like installing node to build packages is very unstable recently (connection reset by peer etc.) - we will get it cached soon WIP is here: https://github.com/apache/airflow/pull/43329 cc: @bugraoz93 that should improve stability.

**Comment by dabla** [comment]
> It seems like installing node to build packages is very unstable recently (connection reset by peer etc.) - we will get it cached soon WIP is here: #43329 cc: @bugraoz93 that should improve stability.

Isn't there a way to be able to re-run a failing step in the ci cd checks of github?

**Comment by dabla** [comment]
> For now rebasing, commit--amend or close/reopening the PR should retry it.

Yeah just did it again, I was a bit worried cause my other PR didn't suffer from those random errors so I though maybe something was wrong in this one.

**Comment by bugraoz93** [comment]
> > For now rebasing, commit--amend or close/reopening the PR should retry it.
> 
> Yeah just did it again, I was a bit worried cause my other PR didn't suffer from those random errors so I though maybe something was wrong in this one.

This is purely too many trials on node servers trying to download the dependencies and appear in random CI runs

File: providers/src/airflow/providers/common/sql/provider.yaml [hunk 1]
```diff
@@ -25,6 +25,7 @@ state: ready
 source-date-epoch: 1730012422
 # note that those versions are maintained by release manager - do not update them manually
 versions:
+  - 1.19.1
```
Comment (eladkal) [inline providers/src/airflow/providers/common/sql/provider.yaml]: Please don't set versions unless explicitly asked by release manager.
It causes troubles during release

Comment (dabla) [reply providers/src/airflow/providers/common/sql/provider.yaml]: Will revert that @eladkal had the same remark from @potiuk my bad

Comment (potiuk) [inline providers/src/airflow/providers/common/sql/provider.yaml]:
```suggestion
```

Comment (potiuk) [reply providers/src/airflow/providers/common/sql/provider.yaml]: I did it via Github UI / suggestions :)

**Review by potiuk — APPROVED** [review]

**Comment by dabla** [comment]
> Nice!

Thx @potiuk

**Pull request merged**